hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/prometheus/RestTemplateParallelPrometheusClient.java

```diff
@@ -0,0 +1,114 @@
+package pl.allegro.tech.hermes.management.infrastructure.prometheus;
+
+import com.google.common.base.Preconditions;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.client.RestTemplate;
+import pl.allegro.tech.hermes.api.MetricDecimalValue;
+import pl.allegro.tech.hermes.management.infrastructure.metrics.MonitoringMetricsContainer;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.net.URLEncoder.encode;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+
+public class RestTemplateParallelPrometheusClient implements PrometheusClient {
```

Comment:
nit: i'd consider dropping the `parallel` from class name and configuration options, i .e.  we can use 
`RestTemplatePrometheusClient` and `fetchTimeout`. In `ExternalMonitoringClientProperties` we would use

```
    private int fetchTimeoutMillis = 5000;
    private int fetchingThreads = 30;
```

not a big of `parallel`. These tasks will not necessarily run in parallel (they will run concurrently), but i'd not say this is so important the we should mention it everywhere, i.e. we dont do that with other pieces of Hermes code that are concurrent